chromedp_test.go

```diff
@@ -716,3 +718,64 @@ func TestGracefulBrowserShutdown(t *testing.T) {
 		}
 	}
 }
+
+func TestAttachingToWorkers(t *testing.T) {
+	mux := http.NewServeMux()
+	mux.Handle("/", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "text/html")
+		io.WriteString(w, `<html>
+      <body>
+        <script>
+          new Worker('/worker.js');
+        </script>
+      </body>
+    </html>`)
+	}))
+	mux.Handle("/worker.js", http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "text/javascript")
+		io.WriteString(w, "console.log('I am worker code.');")
+	}))
+	ts := httptest.NewServer(mux)
+	defer ts.Close()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	ctx, cancel = NewContext(context.Background())
+	defer cancel()
+
+	ch := make(chan error, 1)
+
+	ListenTarget(ctx, func(ev interface{}) {
+		if ev, ok := ev.(*target.EventAttachedToTarget); ok {
+			if ev.TargetInfo.Type != "worker" {
+				return
+			}
+			go func() {
```

Comment:
hm, when I suggested the use of a channel, I was thinking more like:

```
ListenTarget(...) {
    ch <- someValue
}
Run(Navigate(...))
value := <-ch
Run(...)
```

That is, keep all the actions in the main goroutine, to keep the flow of actions linear. You also don't need the extra goroutine, since your channel is buffered.

Reply:
Done.